packit/cli/validate_config.py

```diff
@@ -0,0 +1,52 @@
+# MIT License
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Validate PackageConfig
+"""
+
+import logging
+import os
+
+import click
+
+from packit.cli.types import LocalProjectParameter
+from packit.cli.utils import cover_packit_exception
+from packit.cli.utils import get_packit_api
+from packit.config import pass_config, get_context_settings
+from packit.config.package_config import PackageConfigValidation
+
+logger = logging.getLogger(__file__)
+
+
+@click.command("validate-config", context_settings=get_context_settings())
+@click.argument("path_or_url", type=LocalProjectParameter(), default=os.path.curdir)
+@pass_config
+@cover_packit_exception
+def validate_config(config, path_or_url):
+    """
+    Validate PackageConfig - check if specfile_path and synced_files are present and
+    if downstream_package_name is present in Fedora
+    """
+    api = get_packit_api(config=config, local_project=path_or_url)
+    validation = PackageConfigValidation(api.package_config)
```

Comment:
I find doing all the work in the constructor very confusing and out of pattern for how Python code is usually written. Other commands seem to call the API at this point. It would make sense to see something similar here too.